protocol/0014-ORDT-order_types.md

```diff
@@ -46,9 +46,129 @@ Notes on scope of current version of this spec:
+
+* Icebergs may be post only.
+
+* `initial peak size` must be greater than or equal to minimum position size (i.e. minimum order size).
+
+* `minimum peak size` must be `>` 0 and `≤ initial peak size`
+
+
+#### Execution and subsequent refresh
+
+* On entry, if an iceberg order is crossed with the best bid/ask, it trades first with its **full quantity**, i.e. the peak sizes do not come into play during aggressive execution.
+This is to prevent an iceberg order ever being crossed after refreshing.
+
+* Once they enter the book passively, iceberg orders trade just like non-iceberg persistent order, as if the order entered the book with `quantity = initial peak size` on submission, and again each time they are refreshed until `remaining == 0` (or they are cancelled or expired, etc.).
+That is:
+
+  * On entry, unlike normal orders, `displayed quantity` is set to `initial peak size` not `quantity`.
+
+  * As for any other order, `remaining == quantity` on entry.
+
+* When an iceberg order trades, both `remaining` and `displayed quantity` are reduced by the trade size.
+
+* Iceberg orders can trade many times without refresh, reducing `displayed quantity` each time.
+The order will not be refreshed after each trade while `displayed quantity ≥ minimum peak size`.
+
+* When `displayed quantity < minimum peak size` and `remaining > displayed quantity` the order will be refreshed:
+
+  * The refresh happens at the end of the transaction when the order becomes eligible for refresh.
+
+  * On refresh `display quantity` is set to `min(remaining, initial peak size)`.
+
+  * A refresh simulates a cancel/replace, which means that on refresh an iceberg order will always lose time priority relative to other orders at the same price.
+
+  * If multiple iceberg orders need to be refreshed at the same time, they are refreshed in the order that their eligibility for refresh was triggered, so the iceberg that dropped below its `minimum peak size` first is refreshed first (even during the same transaction the sequence of execution must be respected).
+
+* Once the remaining quantity is equal to the displayed quantity, no further refresh is possible.
+The order now behaves like a normal limit order and will leave the book if it trades away completely.
+
+* For an incoming order with size larger than the total displayed quantity at a given price level, the following procedure should be followed:
+  
+  * The incoming order trades with all visible volume at the price level, whether an iceberg order or a vanilla limit order
+
+  * If there is still remaining volume in the order once all visible volume at the price level is used up, the remaining quantity is split between the non-visible components of all iceberg orders at this level according to their remaining volumes. For example if there are three iceberg orders with remaining volume 200 lots, 100 lots and 100 lots, an order for 300 lots would be split 150 to the first order and 75 to the two 100 lot orders. 
```

Comment:
Any preference on where we put the crumbs in the case where the distribution doesn't divide exactly? For example there is 100 left in the aggressive order and 3 icebergs with all equal remaining hidden volume so it'll be 333 each. Just stick the extra 1 onto the iceberg which is first in terms of time priority?